src/zinoargus/__init__.py

```diff
@@ -277,6 +278,47 @@ def synchronize_continuously(argus_incidents: IncidentMap, zino_cases: CaseMap):
             synchronize_case_history(case, incident)
 
 
+def refresh_argus_incidents(argus_incidents: IncidentMap, zino_cases: CaseMap):
+    """Refreshes all the Argus incidents that we know of and care about.
+
+    This also triggers actions that modify Zino cases if indicated by changes to the
+    Argus incidents.
+    """
+    for case_id, old_incident in argus_incidents.items():
+        new_incident = _argus.get_incident(old_incident.pk)
+        argus_incidents[case_id] = new_incident
+
+        if new_incident.acked and not old_incident.acked:
+            update_case_acknowledged(case_id, argus_incidents, zino_cases)
+
+
+def update_case_acknowledged(
+    case_id: int, argus_incidents: IncidentMap, zino_cases: CaseMap
+):
+    """Updates a Zino case with the acknowledged status from Argus, if necessary."""
+    incident = argus_incidents[case_id]
+    msg = "Argus incident %s (zino case %s) was acknowledged"
+    desired_state = _config.sync.acknowledge.setstate
+    if desired_state == "none":
+        _logger.debug(msg + ", ignoring as configured", incident.pk, case_id)
+        return
+
+    case = zino_cases[case_id]
+    if case.state == desired_state:
+        return
+
+    _logger.info(msg + ", setting zino case to %r", incident.pk, case_id, desired_state)
+
+    acks = _argus.get_incident_acknowledgements(incident)
+    _logger.debug("Argus incident %s acks: %r", incident.pk, acks)
+    acks.sort(key=lambda ack: ack.event.timestamp, reverse=True)
```

Comment:
Might be nice to move this before the log message, then we have it sorted there as well

Reply:
Good idea.